### DISCUSSION TITLE
Where is the code for the XRViewer app?

### DISCUSSION THREAD

**Discussion by camjackson** [mozilla-mobile/firefox-ios#10811] [Q&A] [answered]
I'm trying to find the source code for the iOS WebXR Viewer: https://apps.apple.com/us/app/webxr-viewer/id1295998056

This is the old repo for it, which is now defunct: https://github.com/mozilla-mobile/webxr-ios

In the readme it says:

> This is the code for Version 1 of the WebXR Viewer, and is no longer used. Version 2 of the WebXR Viewer is based on the [Firefox for iOS source](https://github.com/mozilla-mobile/firefox-ios), located in the "webxr" branch for the initial release.

That's pointing to this repo, but there is no such webxr branch here: https://github.com/mozilla-mobile/firefox-ios/branches/all?query=xr

I've tried searching through the code, branches, issues, and PRs on this repo to find where the code is for the WebXR Viewer but I can't seem to find it. Where is it?

For context, the reason I want to find it is so that I can open a PR to add `NSLocalNetworkUsageDescription` to it. Without that I can't test XR apps that I'm developing on my local network, which is a big deal! I already [opened a PR](https://github.com/mozilla-mobile/webxr-ios/pull/196) to do that on the old repo but that's a bit pointless I think.

**Comment by lmarceau** [comment, marked as answer]
@camjackson Found the branch, based on https://github.com/mozilla-mobile/firefox-ios/pull/6943. It's on mozilla's remote again [here](https://github.com/mozilla-mobile/firefox-ios/tree/webxr). The branch is really old and might need some update to be up to date with `main` thought. Let me know if there's anything else we can do!

**Reply by camjackson** [reply]
Thanks @lmarceau that's really helpful! I'd really like to add the `NSLocalNetworkUsageDescription` permission to the WebXR Viewer app so that it can be used to test websites running on your local network. If I open a PR that adds this permission to `Client/Info.plist` on the `webxr` branch, will that do it?

**Reply by camjackson** [reply]
I've opened a PR, I hope I did it right! https://github.com/mozilla-mobile/firefox-ios/pull/10826
